helix-core/src/main/java/org/apache/helix/messaging/handling/HelixTaskExecutor.java

```diff
@@ -630,6 +634,20 @@ void unregisterMessageHandlerFactory(String type) {
             + ", pool: " + pool);
   }
 
+  private void syncFactoryState() {
+    for (Map.Entry<String, MsgHandlerFactoryRegistryItem> entry : _hdlrFtyRegistry.entrySet()) {
```

Comment:
Sorry for the confusing. My bad. This factory registration is still needed to be synchronized. Because the following function reset could be parallel called.